framework/Optimizers/GradientDescent.py

```diff
@@ -526,7 +526,13 @@ def _submitOptAndGrads(self, opt, traj, step, stepSize):
     self._submitRun(opt, traj, step, 'opt')
     # GRAD POINTS
     # collect grad points
-    gradPoints, gradInfos = self._gradientInstance.chooseEvaluationPoints(opt, stepSize)
+    # HACK FIXME TODO adding constraints too
+    ## boundary is distribution dict of bounds
+    ## constraint functions is list of functions to call "evaluate" on
+    ## inputs is dictionary of other stuff that constraints might need to be evaluated
+    ## TODO we really should pass the checkFunctionalConstraints and check/applyBoundaryConstraints instead!!
+    constraints = {'boundary': self.distDict, 'functional': self._constraintFunctions, 'inputs': copy.deepcopy(self.constants), 'normalize': self.normalizeData, 'denormalize': self.denormalizeData}
```

Comment:
can you open an issue associated to this?

Reply:
already did, #1412